lib/versionmanager.js

```diff
@@ -241,12 +242,28 @@ function upgradePackageDefinitions(currentDependencies, options) {
  * @param {Object} [options={}]
  * @returns {string} The updated package data, as utf8 text
  */
-function upgradePackageData(pkgData, oldDependencies, newDependencies, newVersions, options = {}) {
+async function upgradePackageData(pkgData, oldDependencies, newDependencies, newVersions, options = {}) {
     for (const dependency in newDependencies) {
         if (!options.minimal || !isSatisfied(newVersions[dependency], oldDependencies[dependency])) {
+            if (options.upgrade && options.interactive) {
```

Comment:
Let's remove the `options.upgrade` requirement. It will still work with it but it can be treated independently.